src/Controller/UI/NationalDataCenterController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Controller\UI;
+
+use App\Form\NationalDataCenterType;
+
+use App\Entity\NationalDataCenter;
+
+use Symfony\Component\Routing\Annotation\Route;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * The National Data Center Controller class.
+ */
+class NationalDataCenterController extends AbstractController
+{
+    /**
+     * The default action for National Data center.
+     *
+     * @param string $id The id of the national data center.
+     *
+     * @Route("/national-data-center/{id}", name="pelagos_app_ui_nationaldatacenter_default")
+     *
+     * @throws NotFoundHttpException When no national data center is found with this id.
+     *
+     * @return Response A Response instance.
+     */
+    public function defaultAction($id = null)
+    {
+        // Checks authorization of users
+        if (!$this->isGranted('ROLE_DATA_REPOSITORY_MANAGER')) {
+            return $this->render('template/AdminOnly.html.twig');
+        }
+
+        if (!empty($id)) {
+            $nationalDataCenter = $this->entityHandler->get(NationalDataCenter::class, $id);
+            if (!$nationalDataCenter instanceof NationalDataCenter) {
+                throw $this->createNotFoundException('The National data center was not found');
+            }
+        } else {
+            $nationalDataCenter = new NationalDataCenter();
+        }
+
+        $form = $this->get('form.factory')->createNamed(null, NationalDataCenterType::class, $nationalDataCenter);
+
+        return $this->render('template/NationalDataCenter.html.twig', array(
+            'NationalDataCenter' => $nationalDataCenter,
+            'form' => $form->createView(),
+            'entityService' => $this->entityHandler
```

Comment:
This entityhandler is not freely available anymore. Also whoemever passed the entityhandler inside the template should be ashamed of themselves.

Reply:
I don't know why it is passed from all the Controllers, it is being used only in researchGroup.html.twig. I'll remove them and refactor it